server/src-lib/Hasura/GraphQL/Schema.hs

```diff
@@ -666,6 +738,19 @@ mkInsInpTy :: QualifiedTable -> G.NamedType
 mkInsInpTy tn =
   G.NamedType $ qualTableToName tn <> "_insert_input"
 
+-- table_relname_obj_insert_input
+mkObjInsInpTy :: QualifiedTable -> RelName -> G.NamedType
```

Comment:
These types don't need to be generated per (table, rel). They can be as follows:

```graphql
input table_insert_input {
  ..
}
input table_obj_rel_insert_input {
  data table_insert_input!
  on_conflict table_on_conflict!
}
input table_arr_rel_insert_input {
  data [table_insert_input!]!
  on_conflict table_on_conflict!
}
```